tripal/tripal
# Fix for remote__data field.

Issue #423

## Type(s) of Change(s)


## Description


As reported in issue #423 the remote__data field was not properly working. The problem was deeper than the error messages reported in the issue.  This  PR contains a variety of fixes that make the field work.  

## Testing?




This test made a fix to the web services so to test this both the remote and local sites need to be on this branch.  An easy workaround is to just use the local site's web services as if it were a remote site.  Follow these steps to test:
1)  Pull the branch for this PR
2)  Make sure the tripal_ws module is enabled.
3)  Add a new Tripal remote site that really points to the local site.  Do this by going to Tripal > Data Storage > Remote Tripal Sites > Configuration.  For my local test machine I used the URL http://localhost and version 'v0.1'.
4)  Go to the Organism content type:  Structure > Tripal Content Types and click manage fields on the Organism content type and create a new field of type 'Remote Tripal Site'.  Set the Remote Tripal Site to the site you created in step 3;  set the query to "Organism?genus=[taxrank__genus]&species=[taxrank__species]";  set the Field to display to 'abbreviation' or if your organisms don't have an abbreviation set it to 'genus' or 'species'. If you have used the NCBI taxonomy importer then you can use 'division' or any other field added by that importer.  For the test, the Controlled Vocabulary Term at th end doesn't matter. Just pick something that doesn't already belong to a field for the content type.
5)  Make sure the new field shows up on the page by clicking on the 'Manage Display' tab at the top and moving it out of the Disabled section.
6)  Go to any organism page that already exists on the site.  You should see the new field that houses data from the "remote site" (actually your local site just repeated).   If you received no errors along the way and the data is shown then all is good.

## Screenshots (if appropriate):

## Additional Notes (if any):
